pkg/endpoint/policy.go

```diff
@@ -545,6 +525,14 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 		return e.nextPolicyRevision > e.policyRevision, nil
 	}
 
+	e.prevIdentityCache = labelsMap
+
+	// First step when calculating policy is to check whether ingress or egress
+	// policy applies (i.e., if rules select this endpoint). We can use this
+	// information to short-circuit policy generation if enforcement is
+	// disabled for ingress and / or egress.
+	e.ingressPolicyEnabled, e.egressPolicyEnabled = owner.EnableEndpointPolicyEnforcement(e)
```

Comment:
I have another PR coming up soon which checks these values before generating endpoint policy. We can short-circuit policy-generation in this case.